### DISCUSSION TITLE
model requires reload to see previous changes even though element factory persists

### DISCUSSION THREAD

**Discussion by mikekidner** [gaphor/gaphor#3591] [Q&A]
Hi, I'm writing a script to create test case activities based on requirements. like this:

```
 if args.create:
       # 1st block: create package for test cases
        with Transaction(event_manager):
            element_factory = create_test_case_package(element_factory)
            with open(model, "w", encoding="utf-8") as out:
                storage.save(out, element_factory)

        #Reload the model here to force above changes to be recognised.
        element_factory, _, event_manager = load_model(model)

       # 2nd block: Create test cases in the new package 
        with Transaction(event_manager):
            element_factory = create_test_cases(element_factory)
            with open(model, "w", encoding="utf-8") as out:
                storage.save(out, element_factory)
```

I've noticed that I've had to reload the model to get the first set of changes to be recognised in the second block. Without this I end up with an empty package from the first block, and then on running the code again an additional instance of the package that contains the testcase elements as intended. 

Is this element factory behaviour expected?

**Comment by amolenaar** [comment]
Where did you run this script? In the interactive console or a separate script?

**Reply by mikekidner** [reply]
I run this as a separate script.

**Reply by amolenaar** [reply]
I'm not sure what's happening. Is it possible to share the complete script?

**Reply by mikekidner** [reply]
@amolenaar , you can find the script and the gaphor file I was testing on in the gaphor_tools repo here:

https://bitbucket.org/resonatesystems/gaphor_tools/src/GAP-27-Automagic-testcases/src/gaphor_tools/scripts/generate_test_cases.py

https://bitbucket.org/resonatesystems/gaphor_tools/src/GAP-27-Automagic-testcases/src/gaphor_tools/scripts/tc.gaphor

Let me know if you have any issues accessing these.
